### ISSUE TITLE
keycloak ingress 502 errors

### ISSUE THREAD

**Issue by jack1902** [bitnami/charts#9043]
### Name and Version

bitnami/keycloak 6.1.5

### What steps will reproduce the bug?

1. Install keycloak using the `values-ha.yaml` in the `ci` folder onto a cluster with `ingress-nginx`
2. Login to the `admin` auth panel
3. hit refresh a couple of times to observe a 502

### What is the expected behavior?

Things to "just-work" out of the box when using the `ha` values

### What do you see instead?

a 502 error from `ingress-nginx` (due to a lack of configuration or knowledge upfront on the keycloak header sizes)

Check `ingress-nginx` logs and see:

```log
2022/02/16 16:03:43 [error] 2718#2718: *645594 upstream sent too big header while reading response header from upstream, client: REDACTED, server: REDACTED, request: "GET /auth/realms/master/protocol/openid-connect/<TRUNCATED_FOR_VISIBILITY>", host: "REDACTED"
```

### Additional information

This is more of a "by the way" for anyone else who might end up using the `bitnami` keycloak chart and encounter a similar error.

I fixed this following some information from the following sources:

https://www.ibm.com/support/pages/502-error-ingress-keycloak-response
https://stackoverflow.com/questions/56126864/why-do-i-get-502-when-trying-to-authenticate
https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#proxy-buffer-size

TL;DR

If using ingress-nginx, add the following annotation to see a fix:

```yaml
annotations:
  nginx.ingress.kubernetes.io/proxy-buffer-size: "128k"
```

Adding the below for better search finds in the "issues":
502 500 ingress keycloak admin

**Comment by jack1902** [comment]
Unsure of how to address this from within the `readme` without polluting it for all example ingress use-cases since I am unsure if this same error happens on other ingresses with keycloak

**Comment by javsalgar** [comment]
Hi!

Thank you for sharing this. It would be great if other ingress users (non-nginx) could share their findings on this. Maybe a note right now for the nginx ingress controller case should suffice. Would you like to contribute with a PR?

**Comment by jack1902** [comment]
Sure, perhaps in the readme? Nginx "gotchas" or something similar?

**Comment by javsalgar** [comment]
In the Ingress section could work, yes

**Comment by nkreiger** [comment]
@jack1902 can confirm, that it does happen to others :) Thanks for the fix.
